src/GrammarParser.php

```diff
@@ -138,35 +160,34 @@ public function buildGrammar($grammarStr, $options = [])
         return $grammar;
     }
 
-    public function getParser()
+    public function getParser(): Parser
     {
         if ($this->parserSchouldBeRefreshed) {
-            $this->generateNewParser();
+            $this->parser = $this->generateNewParser();
             $this->parserSchouldBeRefreshed = false;
         }
 
         return $this->parser;
     }
 
-    protected function buildBranchNameNode()
+    protected function buildBranchNameNode(): GrammarItemRestrictions
     {
         $restrictedWords = ['or', 'and', 'contain', 'is', 'text', 'string'];
 
         $restrictedWordsGrammarNode = [];
         foreach ($restrictedWords as $restrictedWord) {
-            $restrictedWordsGrammarNode[] = new \ParserGenerator\Extension\ItemRestrictions\Is(new \ParserGenerator\GrammarNode\TextS($restrictedWord));
+            $restrictedWordsGrammarNode[] = new Is(new TextS($restrictedWord));
         }
 
-        $q = new \ParserGenerator\GrammarNode\ItemRestrictions(
-            new \ParserGenerator\GrammarNode\Regex('/[A-Za-z_][0-9A-Za-z_]*/', true),
-            new \ParserGenerator\Extension\ItemRestrictions\ItemRestrictionNot(
-                new \ParserGenerator\Extension\ItemRestrictions\ItemRestrictionOr($restrictedWordsGrammarNode)
+        return
+            new GrammarItemRestrictions(
+                new GrammarRegex('/[A-Za-z_][0-9A-Za-z_]*/', true),
+                new ItemRestrictionNot(
+                    new ItemRestrictionOr($restrictedWordsGrammarNode)
```

Comment:
I ❤️ how much more readable this got…

Reply:
Yep. I have no idea what kind of madnes made me use full class name.